spotbugs-tests/src/test/java/edu/umd/cs/findbugs/SuppressionMatcherMockedTest.java

```diff
@@ -0,0 +1,69 @@
+package edu.umd.cs.findbugs;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class SuppressionMatcherMockedTest {
+
+    @Mock
+    private ClassAnnotation classAnnotation;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private ClassWarningSuppressor suppressor;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private BugInstance bugInstance;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private BugInstance uselessSuppressionBugInstance;
+
+    @Mock
+    private BugReporter bugReporter;
+
+    @BeforeEach
+    void init() {
+        when(suppressor.getClassAnnotation().getTopLevelClass()).thenReturn(classAnnotation);
+        when(bugInstance.getPrimaryClass().getTopLevelClass()).thenReturn(classAnnotation);
+    }
+
+    @Test
+    void shouldMatchBugInstance() {
+        // given
+        when(suppressor.match(bugInstance)).thenReturn(true);
+        SuppressionMatcher matcher = new SuppressionMatcher();
+        // when
+        matcher.addSuppressor(suppressor);
+        // then
+        assertTrue(matcher.match(bugInstance), "Should match BugInstance");
+        // and when
+        matcher.validateSuppressionUsage(bugReporter);
+        // then
+        verifyNoInteractions(bugReporter);
+    }
+
+    @Test
+    void shouldNotMatchBugInstance() {
+        // given
+        when(suppressor.match(bugInstance)).thenReturn(false);
+        when(suppressor.buildUselessSuppressionBugInstance()).thenReturn(uselessSuppressionBugInstance);
```

Comment:
Should this be in `init`?

Reply:
The two tests mock objects differently:
`when(suppressor.match(bugInstance)).thenReturn(true);`
and:
`when(suppressor.match(bugInstance)).thenReturn(false);`

So I think this can't be factored in `init`

Reply:
Those lines are different, yes, but making the suppressor return the instance is the same in both.

Reply:
Line 57, specifically, is duplicated across multiple tests, and could be extracted to the setup method.

    when(suppressor.buildUselessSuppressionBugInstance()).thenReturn(uselessSuppressionBugInstance);

Reply:
Sorry I'm still lost, I don't see duplications for that line in `SuppressionMatcherMockedTest`